apiserver/apiserver.go

```diff
@@ -504,6 +489,32 @@ func (w logsinkMetricsCollectorWrapper) LogReadCount(modelUUID, state string) pr
 	return w.collector.LogReadCount.WithLabelValues(modelUUID, state)
 }
 
+// httpRequestRecorderWrapper defines a wrapper from exposing the
+// essentials for the http request recorder.
+type httpRequestRecorderWrapper struct {
+	collector *Collector
+	modelUUID string
+}
+
+// Record an outgoing request which produced an http.Response.
+func (w httpRequestRecorderWrapper) Record(method string, url *url.URL, res *http.Response, rtt time.Duration) {
+	// Note: Do not log url.Path as REST queries _can_ include the name of the
+	// entities (charms, architectures, etc).
+	w.collector.TotalRequests.WithLabelValues(w.modelUUID, url.Host, strconv.FormatInt(int64(res.StatusCode), 10)).Inc()
```

Comment:
I would make helpers for these - uniform access principle. Also means you don't refer to multiple members in a call.

Reply:
Generally agree but in this case (just two call sites) it seems to me like overkill to add helpers.